### DISCUSSION TITLE
How to use GET and POST action

### DISCUSSION THREAD

**Discussion by kaduwaengertner** [WarmUpTill/SceneSwitcher#524] [Q&A]
Wondering if anyone get lucky using the GET and POST new actions, not sure if theres anything about it

**Comment by nacalai** [comment]
Well what do you mean by that? They work as they intend to

tir. 6. sep. 2022, 21:25 skrev Kadu Waengertner ***@***.***>:

**Reply by isopigeon** [reply]
Where does the data go, looking at the log i can see the request being sent, but where is the reply? (for GET)
and for POST , what is the formatting? i assume these are being sent using CURL? is the text box just the Data portion of the request or does it allow for headers as well?

If you could point me to any data about the Http action it would be appreciated.

**Reply by WarmUpTill** [reply]
> where is the reply?

The reply data is being discarded.
Note that if you want to use GET requests' response data in a condition you can use the "File" condition type's option to check the content of "remote" files, as it also is simply using CURL to send a GET request.
(I might have to clean this up in future and move this functionality to a "Http condition" of sorts)

>i assume these are being sent using CURL?

Correct.

>is the text box just the Data portion of the request or does it allow for headers as well?

Just the data.

> If you could point me to any data about the Http action it would be appreciated.

Unfortunately there is no documentation yet, but I will try to answer any questions.
(And afterwards add an entry to the wiki)

I case you are looking for the source code for the GET and POST functions of the Http action you can find them here:

https://github.com/WarmUpTill/SceneSwitcher/blob/bddbcebece6d7b12472530fdc3f8ffbe64faf289/src/macro-core/macro-action-http.cpp#L25-L44
